lib/matplotlib/tight_layout.py

```diff
@@ -34,40 +34,33 @@ def _get_top(tight_bbox, axes_bbox):
     return tight_bbox.ymax - axes_bbox.ymax
 
 
-def auto_adjust_subplotpars(fig, renderer,
-                            nrows_ncols,
-                            num1num2_list,
-                            subplot_list,
-                            ax_bbox_list=None,
-                            pad=1.08, h_pad=None, w_pad=None,
-                            rect=None):
+def auto_adjust_subplotpars(
+        fig, renderer, nrows_ncols, num1num2_list, subplot_list,
+        ax_bbox_list=None, pad=1.08, h_pad=None, w_pad=None, rect=None):
     """
-    Return a dictionary of subplot parameters so that spacing between
-    subplots are adjusted. Note that this function ignore geometry
-    information of subplot itself, but uses what is given by
-    *nrows_ncols* and *num1num2_list* parameteres. Also, the results could be
-    incorrect if some subplots have ``adjustable=datalim``.
-
-    Parameters:
-
-    nrows_ncols
-      number of rows and number of columns of the grid.
-
-    num1num2_list
-      list of numbers specifying the area occupied by the subplot
-
-    subplot_list
-      list of subplots that will be used to calcuate optimal subplot_params.
-
+    Return a dict of subplot parameters to adjust spacing between subplots.
+
+    Note that this function ignores geometry information of subplot
+    itself, but uses what is given by the *nrows_ncols* and *num1num2_list*
+    parameters.  Also, the results could be incorrect if some subplots have
+    ``adjustable=datalim``.
+
+    Parameters
+    ----------
+    nrows_ncols : Tuple[int, int]
+        Number of rows and number of columns of the grid.
+    num1num2_list : List[int]
```

Comment:
Is this a standard way of documenting lists of ints (for example)? It looks new to me, and if it is the standard it should be in our documentation guide to avoid confusion.

Reply:
Suggested in https://github.com/matplotlib/matplotlib/issues/9271 ("typehints-style"), and OK'd by @tacaswell further down the comments.
Will edit the style guide in a separate PR.